enhancements/ingress/default-ingress-controller.md

```diff
@@ -0,0 +1,110 @@
+
+This proposal eliminates the contention between the operator and administrators.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This proposal eliminates the operator/administrator contention by building on the [User Defined IngressControllers](user-defined-default-ingress-controller.md) capability. Specifically, by:
+
+1. Changing the ingress-operator to stop creating ingresscontrollers altogether
+2. Changing the installer to always render a `default` IngressController
+
+These changes make explicit the notion that the ingress-operator is responsible only for managing IngressControllers created by end users, _not_ for creating IngressControllers.
+
+End users can now assume that the _only_ way IngressControllers are created are via the installer or through their own API calls.
+
+## Motivation
+
+The operator today has a confusing relationship to the `default` IngressController resource. Not only does the operator confound automation in its competition with users for ownership, but the operator is not even guaranteed to have the context necessary to restore any IngressController that was deleted, including the `default` instance. The operator should have a clear responsibility to manage resources brought by the user — the installer being indistinguishable from a user in this context. The installer should have a clear responsibility to always bring the `default` IngressController resource.
+
+### Goals
+
+1. Eliminate contention with users/automation trying to replace the default IngressController post-installation.
+2. Clarify the role of the ingress-operator with respect to IngressController resource creation
+3. Clarify the role of the installer with respect to the default IngressController resource
+
+### Non-Goals
+
+
+## Proposal
+
+The proposal spans two components: the ingress-operator, and the installer.
+
+The ingress-operator should be modified to stop creating the `default` IngressController resource.
+
+Currently, the installer only renders the `default` IngressController resource when the cluster is configured as "internal". The installer should be changed to _always_ render the `default` IngressController.
```

Comment:
>so the work of deciding what the default ingress controller should look like for various platforms now moves from the operator to the installer.

I'm not proposing that, so maybe we should talk more in depth about the assumption.

Today, if the cluster is configured as "private", the installer renders a default ingresscontroller with the minimal schema to allow the operator to fill in the blanks based on cluster config and come out the other side with an internal default. This is actually more decision-making already present than what I'm proposing here.

The proposal is that in addition to the above, when the cluster is public (the default), the installer renders _the minimal ingresscontroller_ definition to allow the operator to do the right thing, _not_ for the installer to make any other specific decisions. So, the installer would render an ingresscontroller with, literally, an empty spec.

Everything else you said seems to be predicated on that assumption, so let's see if we can come to an understanding on this point first...

Reply:
> The proposal is that in addition to the above, when the cluster is public (the default), the installer renders the minimal ingresscontroller definition to allow the operator to do the right thing, not for the installer to make any other specific decisions. So, the installer would render an ingresscontroller with, literally, an empty spec.

the assumption that allow ^^ seems fine to me. I just want to prevent any platform based decision making in the installer..

Reply:
Thanks... I updated the EP to try and make this more clear